test/unit/BFactory.t.sol

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.23;
+
+import {BFactory} from 'contracts/BFactory.sol';
+import {BPool} from 'contracts/BPool.sol';
+import {IERC20} from 'contracts/BToken.sol';
+import {Test} from 'forge-std/Test.sol';
+
+abstract contract Base is Test {
+  BFactory public bFactory;
+  address public owner = makeAddr('owner');
+
+  function setUp() public {
+    vm.prank(owner);
+    bFactory = new BFactory();
+  }
+}
+
+contract BFactory_Unit_Constructor is Base {
+  /**
+   * @notice Test that the owner is set correctly
+   */
+  function test_correctDeploy() public view {
+    assertEq(owner, bFactory.getBLabs());
+  }
+}
+
+contract BFactory_Unit_IsBPool is Base {
+  /**
+   * @notice Test that a valid pool is present on the mapping
+   */
+  function test_validPoolIsBPool() public {
+    BPool _pool = bFactory.newBPool();
```

Comment:
can we avoid deploying a new pool but instead overwriting the storage with cheatcodes? 🙏